app/commands/git/sync_track.rb

```diff
@@ -0,0 +1,86 @@
+module Git
+  class SyncTrack < Sync
+    include Mandate
+
+    def initialize(track)
+      super(track, track.synced_to_git_sha)
+      @track = track
+    end
+
+    private
+    attr_reader :track
+
+    def sync!
+      return track.update!(synced_to_git_sha: head_git_track.commit.oid) unless track_needs_updating?
+
+      # TODO: consider raising error when slug in config is different from track slug
+      # TODO: validate track to prevent invalid track data
+      track.update!(
+        blurb: head_git_track.config[:blurb],
+        active: head_git_track.config[:active],
+        title: head_git_track.config[:language],
+        synced_to_git_sha: head_git_track.commit.oid,
+        concepts: concepts,
+        concept_exercises: concept_exercises,
+        practice_exercises: practice_exercises
+      )
+
+      track.concepts.each { |concept| Git::SyncConcept.(concept) }
+      track.concept_exercises.each { |concept| Git::SyncExercise.(concept) }
+      track.practice_exercises.each { |concept| Git::SyncExercise.(concept) }
+    end
+
+    def concepts
+      # TODO: verify that all exercise concepts and prerequisites are in the concepts section
+      config_concepts.map do |config_concept|
+        ::Track::Concept::Create.(
+          config_concept[:uuid],
+          config_concept[:slug],
+          config_concept[:name],
+          config_concept[:blurb],
+          head_git_track.commit.oid,
+          track
+        )
+      end
+    end
+
+    def concept_exercises
+      config_concept_exercises.map do |exercise|
+        ::ConceptExercise::Create.(
+          exercise[:uuid],
+          exercise[:slug],
+          exercise[:name], # TODO: the DB used title, config.json used name. Consider if we want this
+          find_concepts(exercise[:concepts]),
+          find_concepts(exercise[:prerequisites]),
+          exercise[:deprecated] || false,
+          head_git_track.commit.oid,
+          head_git_track.commit.oid,
+          track
+        )
```

Comment:
So with my other comment, this would be:

```suggestion
        ::ConceptExercise::Create.(
          track,
          exercise[:uuid],
          slug: exercise[:slug],
          name: exercise[:name], # TODO: the DB used title, config.json used name. Consider if we want this
          concepts: find_concepts(exercise[:concepts]),
          prerequisites: find_concepts(exercise[:prerequisites]),
          deprecated: exercise[:deprecated] || false,
          synced_to_git_sha: head_git_track.commit.oid
        )
```